sonic_platform_base/chassis_base.py

```diff
@@ -451,6 +453,95 @@ def get_thermal_manager(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # Voltage Sensor Methods
+    ##############################################
+
+    def get_num_vsensors(self):
+        """
+        Retrieves the number of voltage sensors available on this chassis
+
+        Returns:
+            An integer, the number of voltage sensors available on this chassis
+        """
+        return len(self._vsensor_list)
+
+    def get_all_vsensors(self):
+        """
+        Retrieves all voltage sensors available on this chassis
+
+        Returns:
+            A list of objects derived from VsensorBase representing all voltage 
+            sensors available on this chassis
+        """
+        return self._vsensor_list
+
+    def get_vsensor(self, index):
+        """
+        Retrieves voltage sensor unit represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the voltage sensor to
+            retrieve
+
+        Returns:
+            An object dervied from VsensorBase representing the specified voltage sensor
+        """
+        vsensor = None
+
+        try:
+            vsensor = self._vsensor_list[index]
+        except IndexError:
+            sys.stderr.write("Voltage sensor index {} out of range (0-{})\n".format(
```

Comment:
We should start using f-strings instead of `.format` when possible. For example, this would become:
`f"Voltage sensor index {index} out of range (0-{len(self._vsensor_list)-1})\n"`

Reply:
SONiC is still supporting python 2.7 so until that is deprecated, it might be risky to do that in common code.